engine/src/test/java/io/zeebe/engine/processor/TypedEventSerializationTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.processor;
+
+import static io.zeebe.util.buffer.BufferUtil.wrapString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.zeebe.logstreams.log.LoggedEvent;
+import io.zeebe.protocol.impl.record.RecordMetadata;
+import io.zeebe.protocol.impl.record.value.deployment.DeploymentRecord;
+import io.zeebe.protocol.record.JsonSerializable;
+import io.zeebe.protocol.record.RecordType;
+import io.zeebe.protocol.record.RejectionType;
+import io.zeebe.protocol.record.ValueType;
+import io.zeebe.protocol.record.intent.DeploymentIntent;
+import io.zeebe.protocol.record.value.deployment.ResourceType;
+import io.zeebe.test.util.JsonUtil;
+import org.agrona.DirectBuffer;
+import org.junit.Test;
+
+public class TypedEventSerializationTest {
```

Comment:
Do you think it makes sense to verify that `TypedEvent.toJson()` is equal to `CopiedRecord.toJson()`?
Maybe, we can combine the test with `JsonSerializableToJsonTest`.

Reply:
Comparison of TypedEvent.toJson with CopiedRecord instead of Json String.